docs/articles/persistence/custom-persistence-provider.md

```diff
@@ -0,0 +1,467 @@
+
+##### Handling `AtomicWrite` Result
+
+The journal can also signal that it rejects individual messages (`AtomicWrite`) by the returned `Task`.
+
+* It is possible but not mandatory to reduce the number of allocations by returning null for the happy path, i.e. when no messages are rejected.
+* Otherwise the returned list must have as many elements as the input `messages`.
+* Each result element signals if the corresponding `AtomicWrite` is rejected or not, with an exception describing the problem.
+* Rejecting a message means it was not stored, i.e. it must not be included in a later replay.
+* Rejecting a message is typically done before attempting to store it, e.g. because of serialization error.
+* Data store connection problems must not be signaled as rejections.
+
+##### Code Sample
+
+`InsertEventSql` in this example refers to this SQL query statement:
+
+[!code-csharp[InsertEventSql](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=InsertEventSql "InsertEventSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[WriteMessagesAsync](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=WriteMessagesAsync "WriteMessagesAsync method implementation")]
+
+#### DeleteMessagesToAsync
+
+```c#
+protected abstract Task DeleteMessagesToAsync(
+    string persistenceId, 
+    long toSequenceNr)
+```
+
+* **`persistenceId`**: Persistent actor identifier
+* **`toSequenceNr`**: Inclusive sequence number of the last event to be deleted
+
+This call is protected with a circuit-breaker.
+
+Asynchronously deletes all persistent messages up to inclusive `toSequenceNr` bound. This operation has to be done atomically.
+
+##### Code Sample
+
+`DeleteBatchSql` in this example refers to this SQL query statement:
+
+[!code-csharp[DeleteBatchSql](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=DeleteBatchSql "DeleteBatchSql SQL Statement")]
+
+`HighestSequenceNrSql` in this example refers to this SQL query statement:
+
+[!code-csharp[HighestSequenceNrSql](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=HighestSequenceNrSql "HighestSequenceNrSql SQL Statement")]
+
+`UpdateSequenceNrSql` in this example refers to this SQL query statement:
+
+[!code-csharp[UpdateSequenceNrSql](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=UpdateSequenceNrSql "UpdateSequenceNrSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[DeleteMessagesToAsync](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=DeleteMessagesToAsync "DeleteMessagesToAsync method implementation")]
+
+### Detail Implementation
+
+#### Reading HOCON Settings
+
+There are some HOCON settings that are by default loaded by the journal base class and these can be overriden in your HOCON settings. The minimum HOCON settings that need to be defined for your custom plugin are:
+
+```hocon
```

Comment:
@Arkatufus what is the best option here?

Reply:
Nothing, HOCON is a proprietary format, there are no CSS equivalent